homeassistant/components/nuki/config_flow.py

```diff
@@ -83,6 +86,48 @@ async def async_step_dhcp(self, discovery_info: dict):
 
         return await self.async_step_validate()
 
+    async def async_step_reauth(self, data):
+        """Perform reauth upon an API authentication error."""
+        self._data = data
+
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that inform the user that reauth is required."""
+        errors = {}
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm", data_schema=REAUTH_SCHEMA
+            )
+
+        conf = {
+            CONF_HOST: self._data[CONF_HOST],
+            CONF_PORT: self._data[CONF_PORT],
+            CONF_TOKEN: user_input[CONF_TOKEN],
+        }
+
+        try:
+            info = await validate_input(self.hass, conf)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+
+        if not errors:
+            existing_entry = await self.async_set_unique_id(info["ids"]["hardwareId"])
+            if existing_entry:
+                self.hass.config_entries.async_update_entry(existing_entry, data=conf)
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
```

Comment:
Maybe create a task for this? We don't need to wait for it.
```suggestion
                self.hass.async_create_task(
                    self.hass.config_entries.async_reload(existing_entry.entry_id)
                )
```

Reply:
The pytest is now failing due to this change. Locally it was successfully. I'm not sure here what needs to be fixed / where the problem is, it just states "Event loop is closed"

Reply:
Make sure to block till done before checking the entry state in the test and leaving the test. Ie use `hass.async_block_till_done`. The patch needs to stay in place during the reload, so we need to make sure the reload happens during the test.

Reply:
Thanks, that worked!